src/components/dialog.js

```diff
@@ -140,22 +140,22 @@ export class Dialog {
     }
 
     // Attach.
-    this.doc_.getBody().appendChild(iframe.getElement()); // Fires onload.
-    this.graypane_.attach();
+    return this.doc_.addToFixedLayer(iframe.getElement()).then(() => {
+      this.graypane_.attach();
 
-    if (hidden) {
-      setImportantStyles(iframe.getElement(), {
-        'visibility': 'hidden',
-        'opacity': 0,
+      if (hidden) {
+        setImportantStyles(iframe.getElement(), {
```

Comment:
This branch of `if()` would be one thing I'd recommend moving above `return this.doc_.addToFixedLayer()...`. Otherwise, I'd have a concern that iframe can flash on the screen before hidden here. In other words, let's keep this part of code synchronous.

Reply:
sgtm

Reply:
moot as I had to revert to the old way where it's attached before the xfer to fixed laer - AMP will not take it if it's not attached to the doc.